File: python-service/app/models/video_transcript.py
# ============================================================
# FILE 4: app/models/video_transcript.py
# ============================================================
"""
VideoTranscript Model - AI-generated transcriptions
"""
from sqlalchemy import Column, Integer, String, Text, DateTime, ForeignKey, Numeric
from sqlalchemy.orm import relationship
from sqlalchemy.dialects.postgresql import JSONB
from app.core.database import Base
from datetime import datetime


class VideoTranscript(Base):
    __tablename__ = "video_transcripts"
    __table_args__ = {'schema': 'academic'}

    id = Column(Integer, primary_key=True, index=True)

    # ==================== Relationship ====================
    video_lesson_id = Column(Integer, ForeignKey('academic.video_lessons.id'), nullable=False)
    
    video_lesson = relationship("VideoLesson", back_populates="transcript")

    # ==================== Transcript Data ====================
    full_transcript = Column(Text, nullable=False)
    language = Column(String(10), default='en')
    segments = Column(JSONB)  # Array of {start, end, text}

    # ==================== Metadata ====================
    generated_at = Column(DateTime(timezone=True), default=datetime.utcnow)
    model_used = Column(String(50))
    confidence_score = Column(Numeric(4, 3))

File: java-service/src/main/java/com/edu/platform/model/VideoTranscript.java
package com.edu.platform.model;

import jakarta.persistence.*;
import lombok.*;

import org.hibernate.annotations.JdbcTypeCode;
import org.hibernate.annotations.Type;
import org.hibernate.type.SqlTypes;

import java.math.BigDecimal;
import java.time.Instant;

@Entity
@Table(name = "video_transcripts", schema = "academic")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class VideoTranscript {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    // ==================== Relationship ====================
    
    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "video_lesson_id", nullable = false)
    private VideoLesson videoLesson;

    // ==================== Transcript Data ====================
    
    @Column(name = "full_transcript", nullable = false, columnDefinition = "TEXT")
    private String fullTranscript;

    @Column(name = "language", length = 10)
    @Builder.Default
    private String language = "en";

    // Segments as JSONB (array of {start, end, text})
    @JdbcTypeCode(SqlTypes.JSON)
    @Column(name = "segments", columnDefinition = "jsonb")
    private String segments;

    // ==================== Metadata ====================
    
    @Column(name = "generated_at")
    private Instant generatedAt;

    @Column(name = "model_used", length = 50)
    private String modelUsed;

    @Column(name = "confidence_score", precision = 4, scale = 3)
    private BigDecimal confidenceScore;

    @PrePersist
    protected void onCreate() {
        if (this.generatedAt == null) {
            this.generatedAt = Instant.now();
        }
    }
}